packages/core/src/icon/icon.element.ts

```diff
@@ -35,6 +35,17 @@ import { hasIcon } from './utils/icon.service-helpers.js';
  * <cds-icon shape="user"></cds-icon>
  * ```
  *
+ * In JavaScript, import the icon name from `@clr/core/icon` instead. This also provides type safety when used in TypeScript:
+ *
+ * ```tsx
+ * import { ClarityIcons, userIcon, userIconName } from '@clr/core/icon';
+ * import { CdsIcon } from '@clr/react/icon';
+ *
+ * ClarityIcons.addIcons(userIcon);
+ *
+ * <CdsIcon shape={userIconName} />
+ * ```
+ *
```

Comment:
Let's move this comment block. The one in the MDX file is good however this is React/JSX specific syntax on the custom element that would not apply to all users. We can move this to the React specific wrapper component. You should be able to add a jsdoc comment right above line 6 here

https://github.com/vmware/clarity/blob/next/packages/react/src/icon/index.tsx#L6

Reply:
Done, is that how you meant?